tests/test_regexp.cppo.ml

```diff
@@ -22,6 +22,16 @@ let mkloc = Loc.mkloc
 let mknoloc = Loc.mknoloc
 let map_loc f {Loc.txt = x; loc} = {Loc.txt = f x; loc}
 
+#if OCAML_VERSION < (4, 8, 0)
+let pp_location_error = Location.report_error
+#else
+let pp_location_error = Location.print_report
+#endif
```

Comment:
The `Location` module is snapshot by `ocaml-migrate-parsetree`, you shouldn't need this.

Reply:
This one is more tricky.  I did look for a consistent version of it, but `Migrate_parsetree.Ast_409.Location` is just an alias for the compiler-libs version.  There is a deprecated `Locations` and a `Migrate_parsetree__Locations`.  The latter would work, but it has not been aliased under `Migrate_parsetree`, so I think it should be considered private.

My thoughts on how to handle it:

  1. Make a dummy function and make it simple to replace with the recent version if we need to see the error message. It's just for testing anyway.
  2. Use `Migrate_parsetree__Locations` and find a new solution if it breaks in the future.
  3. Let dune substitute the right version. I think this is no more elegant than using cppo, but it avoids the dependency, if that's the goal.

Do you have other ideas?

Reply:
Oh, it's only for testing! Then either 1 or 3, whichever is easiest to understand/maintain.